GDevelop.js/__tests__/GDJSCodeGenerationIntegrationTests.js

```diff
@@ -602,8 +602,267 @@ describe('libGD.js - GDJS Code Generation integration tests', function () {
+    expect(runtimeScene.getVariables().get('SuccessVariable').getAsNumber()).toBe(5);
+  });
+
+  it('generates a working function with asynchronous actions referring to objects', function () {
+    // Create events using the Trigger Once condition.
+    const eventsSerializerElement = gd.Serializer.fromJSON(
+      JSON.stringify([
+        {
+          disabled: false,
+          folded: false,
+          type: 'BuiltinCommonInstructions::Standard',
+          conditions: [],
+          actions: [
+            {
+              type: { inverted: false, value: 'ModVarObjet' },
+              parameters: ['MyObjectA', 'TestVariable', '+', 'GetArgumentAsNumber("IncreaseValue")'],
+              subInstructions: [],
+            },
+            {
+              type: {
+                inverted: false,
+                value: 'Wait',
+              },
+              parameters: ['1.5'],
+              subInstructions: [],
+            },
+            {
+              type: { inverted: false, value: 'ModVarObjet' },
+              parameters: ['MyObjectA', 'TestVariable', '+', 'GetArgumentAsNumber("IncreaseValue")'],
+              subInstructions: [],
+            },
+          ],
+          events: [],
+        },
+      ])
+    );
+
+    const project = new gd.ProjectHelper.createNewGDJSProject();
+    const eventsFunction = new gd.EventsFunction();
+
+    eventsFunction.getEvents().unserializeFrom(project, eventsSerializerElement);
+
+    const parameter = new gd.ParameterMetadata();
+    parameter.setType('number');
+    parameter.setName('IncreaseValue');
+    eventsFunction.getParameters().push_back(parameter);
+    parameter.setType('object');
+    parameter.setName('MyObjectA');
+    eventsFunction.getParameters().push_back(parameter);
+    parameter.delete();
+
+    const runCompiledEvents = generateCompiledEventsForEventsFunction(
+      gd,
+      project,
+      eventsFunction
+    );
+
+    eventsFunction.delete();
+    project.delete();
+
+    const { gdjs, runtimeScene } = makeMinimalGDJSMock();
+    const myObjectA = runtimeScene.createObject('MyObjectA');
+    const myObjectALists = gdjs.Hashtable.newFrom({ MyObjectA: [myObjectA] });
+    runCompiledEvents(gdjs, runtimeScene, [5, myObjectALists]);
+    expect(myObjectA.getVariables().has('TestVariable')).toBe(true);
+    expect(myObjectA.getVariables().get('TestVariable').getAsNumber()).toBe(5);
+
+    // Process the tasks (but the task is not finished yet).
+    runtimeScene.getAsyncTasksManager().processTasks(runtimeScene);
+    expect(myObjectA.getVariables().get('TestVariable').getAsNumber()).toBe(5);
+
+    // Process the tasks (after faking it's finished).
+    runtimeScene.getAsyncTasksManager().markAllFakeSyncTasksAsFinished();
+    runtimeScene.getAsyncTasksManager().processTasks(runtimeScene);
+    expect(myObjectA.getVariables().get('TestVariable').getAsNumber()).toBe(10);
+  });
+
+  // TODO: Add a test simulating the deletion of an object before a task is finished,
```

Comment:
I feel like this should be done before merging, it's an important test to do

Reply:
Yep, I did add the other tests but ran short of time to make this one. It's super important to have it.